server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java

```diff
@@ -53,22 +53,63 @@ public GradleApiConnector(PreferenceManager preferenceManager) {
   }
 
   /**
-   * Get the Gradle version of the project.
+   * Extracts the GradleVersion for the given project.
+   *
+   * @param projectUri URI of the project to get the gradle version for.
+   * @return Gradle version of the project or empty string upon failure.
    */
   public String getGradleVersion(URI projectUri) {
     try (ProjectConnection connection = getGradleConnector(projectUri).connect()) {
-      return getGradleVersion(connection);
+      return getBuildEnvironment(connection).getGradle().getGradleVersion();
     } catch (BuildException e) {
       LOGGER.severe("Failed to get Gradle version: " + e.getMessage());
       return "";
     }
   }
 
-  private String getGradleVersion(ProjectConnection connection) {
-    BuildEnvironment model = connection
+  /**
+   * Extracts the GradleVersion for the given project connection.
+   *
+   * @param connection ProjectConnection to get the gradle version from.
+   * @return Gradle version of the project or empty string upon failure.
+   */
+  public String getGradleVersion(ProjectConnection connection) {
+    try {
+      return getBuildEnvironment(connection).getGradle().getGradleVersion();
+    } catch (BuildException e) {
+      LOGGER.severe("Failed to get Gradle version: " + e.getMessage());
+      return "";
+    }
+  }
+
+  /**
+   * Extracts the BuildEnvironment model for the given project.
+   *
+   * @param projectUri URI of the project ot get the gradle java home for.
+   * @return BuildEnvironment of the project or {@code null} upon failure.
+   */
+  public BuildEnvironment getBuildEnvironment(URI projectUri) {
+    try (ProjectConnection connection = getGradleConnector(projectUri).connect()) {
+      return connection
```

Comment:
return getBuildEnvironment(connection);

Reply:
Fixed in https://github.com/microsoft/build-server-for-gradle/pull/165/commits/7e4749a704b01421c5b6282ea0204d8856b32636